### ISSUE TITLE
Feature Request: Make bayesplot_grid objects a ggplot object

### ISSUE THREAD

**Issue by billdenney** [stan-dev/bayesplot#268]
Related to #128 and to several other questions (such as https://discourse.mc-stan.org/t/manipulating-bayesplot-plots/4419).

It would be handy if `bayesplot_grid` objects could be modified like any other ggplot object.  I understand that it is not currently a ggplot object is "It wasn't possible (at least it didn't seem possible when we tried) to get the plot how we wanted it to look without creating separate plots and combining them in a grid like that." (according to @jgabry in https://github.com/stan-dev/bayesplot/issues/128#issuecomment-345466831).

But, as I am running into issues much like those in #128 where I want to modify the figure after it is created, it would be helpful to use normal ggplot mechanics to do so.

I'm trying to work around this by using `cowplot::plot_grid()` or similar functions on the `return_value$bayesplot` list of plots.

Is there interest in converting a bayesplot_grid object into something that can be used with normal ggplot2?  (If so, I can try to generalize what I'm working on into a PR.  If not, I won't.)

**Comment by jgabry** [comment]
> Is there interest in converting a bayesplot_grid object into something that can be used with normal ggplot2? (If so, I can try to generalize what I'm working on into a PR. If not, I won't.)

Yeah that would be great! My only concern would be if changing `bayesplot_grid()` like this would break backwards compatibility. But if it would then we could just use a new function name and deprecate `bayesplot_grid()`.
